internal/lefthook/run/prepare_command.go

```diff
@@ -64,8 +66,35 @@ func (r *Runner) prepareCommand(name string, command *config.Command) (*run, err
 	return args, nil
 }
 
+func splitNullTerminatedPaths(s string) []string {
+	var result []string
+	start := 0
+	for i, c := range s {
+		if c == 0 {
+			result = append(result, s[start:i])
+			start = i + 1
+		}
+	}
+	result = append(result, s[start:])
+	return result
+}
+
```

Comment:
I'd like this to be in `internal/lefthook/run.go`, so you parse the files only once and then pass them to all commands as if they were passes via `--files` arg. Otherwise if you have multiple commands using files, only one will receive them.

Reply:
@mrexox My intention was that you had to choose: either from stdin, or `--file` (or deprecated `--files`). This to reduce confusion. And of course, this behavior should be checked at runtime and documented (marking `--files-from-stdin` and `--file(s)` as mutually exclusive). Do you suggest something else here?

Reply:
I'm Ok with this vision. We can return an error if both options get passed. But we could also do the following:

In case user calls lefthook this way: `lefthook run some-hook --files file.ts --files-from-stdin
1. Get the files from `--files` option
2. Parse files from STDIN
3. Concat the results and pass to Runner

This is a more inclusive option and I like it too.

Reply:
That is fine, I'm working on it.

Reply:
Done in https://github.com/evilmartians/lefthook/pull/596/commits/c661ed217df8c58ce4d34e1c992c6169dabb6be3

My previous implementation was not completely correct, since it split on end-of-line characters too. But I'm not a Go programmer and misunderstood the semantics. 

I find it strange that Go uses a `string` type for filesystem operations, that may or may not be a grammatically valid filesystem path. And that this string type isn't even a string but a byte array.